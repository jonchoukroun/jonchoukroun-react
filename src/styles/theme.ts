export default {
    colors: {
        green: "#008080",
        blue: "#000080",
        black: "#141414",
        lightGray: "#f8f8f8",
        gray: "#c0c0c0",
        darkGray: "#878787",
        white: "#fbfbfb",
    },
    header: {
        fontWeight: "500",
        transform: "scaleX(1.1),",
    },
    body: {
        fontSize: "18px",
    },
    border: {
        line: "2px solid",
        color: "#f8f8f8 #7f787f #7f787f #f8f8f8",
        radius: "0",
    },
    defaultDropShadow: "1px 2px 0px 1px rgba(20, 20, 20, 0.7)",
    darkDropShadow: "1px 1px 0px 3px rgba(0, 0, 0, 0.7)",
};
