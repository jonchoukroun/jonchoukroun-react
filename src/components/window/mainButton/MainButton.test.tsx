import React from "react";
import Button from "./MainButton";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import theme from "../../../styles/theme";
import { ThemeProvider } from "styled-components";

function emptyCallback(): Promise<void> {
    return new Promise((resolve) => resolve());
}

const buttonText = "flarg";

it("renders successfully", () => {
    shallow(
        <ThemeProvider theme={theme}>
            <Button
                text={buttonText}
                action={emptyCallback}
                isDisabled={false}
            />
        </ThemeProvider>
    );
});

it("renders the button text", () => {
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <Button text={buttonText} action={emptyCallback} />
        </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a disabled button", () => {
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <Button
                text={buttonText}
                action={emptyCallback}
                isDisabled={true}
            />
        </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("doesn't call the callback when disabled", () => {
    const mock = jest.fn().mockName("mockedCallback");

    const component = shallow(
        <Button text={buttonText} action={mock} isDisabled={true} />
    );
    component.simulate("click");
    expect(mock).not.toHaveBeenCalled();
});
