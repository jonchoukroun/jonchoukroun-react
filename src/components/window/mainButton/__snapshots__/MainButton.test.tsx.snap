// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a disabled button 1`] = `
<button
  className="sc-AxjAm sc-AxirZ iZSVCM"
  disabled={true}
>
  <div
    className="sc-AxiKw sc-AxhCb kPSxiJ"
  >
    <span
      style={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      f
    </span>
    larg
  </div>
</button>
`;

exports[`renders the button text 1`] = `
<button
  className="sc-AxjAm kYJXck"
  onClick={[Function]}
>
  <div
    className="sc-AxiKw iwcMmu"
  >
    <span
      style={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      f
    </span>
    larg
  </div>
</button>
`;
